Re-adds in Assault Ops and Black Mesa Modules

RS-Chan opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

Re-adds in the Assault Ops and Black Mesa Modules for the upcoming sector gameplay.
Reverts the remove changes done back on Main Nova Branch in early January-Feberury 2024. 
Following modules will be placed in modular_nova\modules

## How This Contributes To The BlueMoon Roleplay Experience

Allows further gameplay elements for the upcoming sector where weapons are the main play.

<!-- Please add a short description of why you think these changes would benefit the game and the roleplay atmosphere of the server. If you can't justify it in words, it might not be worth adding. -->

## Proof of Testing

After running in a local server (Pending Changes to the code), confirmed that you are able to purchase Sol Drum Magazines after modifying the Cargo Supply Console Board to "Broad". For balancing issues, currently will keep the standard 60 rounds though can be configured later down the line.

<!-- Include any screenshots/videos/debugging steps of the code functioning successfully, between the </summary> and </details> code blocks. -->
<!-- To our mappers and spriters: Posting screenshots of content INSIDE EDITORS (aseprite, PDN, SDMM, ect) is NOT valid proof of testing. Please make sure that you COMPILE the game and provide PROOF you tested your edits. -->

<details>
<summary>Screenshots/Videos</summary>
  
</details>

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and its effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Re-adds Assault Ops and Black Mesa Modules
code: Re-enables Qarad Drum Magazine for purchase when supply console board is in "Broad" mode 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

RS-Chan committed:
Re-adds in Assault Ops and Black Mesa Modules

Floofies commented:
I would like to fix two issues:

1. Renaming certain people, places, and things to mitigate plagiarism of Microsoft and Sierra Studios intellectual property.

2. I need to verify if the sound effects are licensed correctly and if we're honoring the license. I will explain this one.

I assume the sound effects are directly copied from Black Mesa by Crowbar Collective, which does have some CC BY-NC licensed sound effects in it, but I don't think I see the required things we need to abide by the license terms.

According to the legally binding terms for the CC BY-NC license:
> You must give appropriate credit, provide a link to the license, and indicate if changes were made.
> You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.

RS-Chan commented:
Gotcha, I'll have to take a look into it further then before doing anymore changes/readding in and fulfill the legal requirements.

> I would like to fix two issues:
> 
>     1. Renaming certain people, places, and things to mitigate plagiarism of Microsoft and Sierra Studios intellectual property.
> 
>     2. I need to verify if the sound effects are licensed correctly and if we're honoring the license. I will explain this one.
> 
> 
> I assume the sound effects are directly copied from Black Mesa by Crowbar Collective, which does have some CC BY-NC licensed sound effects in it, but I don't think I see the required things we need to abide by the license terms.
> 
> According to the legally binding terms for the CC BY-NC license:
> 
> > You must give appropriate credit, provide a link to the license, and indicate if changes were made.
> > You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.

RS-Chan commented on modular_nova/modules/black_mesa/code/mobs/gordon_freeman.dm:
Will probably remove/rename

RS-Chan commented on modular_nova/modules/black_mesa/code/mobs/human_mobs.dm:
Rename all units here

RS-Chan commented on modular_nova/modules/black_mesa/code/mobs/nihilanth.dm:
Will either remove or rename in to bring to proper legal requirments

RS-Chan commented on modular_nova/modules/black_mesa/code/mobs/vortigaunt.dm:
Will either remove or rename in to bring to proper legal requirments

RS-Chan commented on modular_nova/modules/black_mesa/code/armaments/_armament_primary.dm:
Rename firearms/replace sounds

RS-Chan commented on modular_nova/modules/black_mesa/code/armaments/_armaments_secondary.dm:
Rename firearms/replace sounds

RS-Chan commented on modular_nova/modules/assault_operatives/code/assault_operatives.dm:
Rename roles/Replace

RS-Chan commented on modular_nova/modules/assault_operatives/code/assault_operatives_outfits.dm:
Verify sprites to legal requirements

RS-Chan commented on modular_nova/modules/assault_operatives/code/interrogator.dm:
> "Bell, we got a job to do"

RS-Chan commented:
> ## About The Pull Request
> 
> Re-adds in the Assault Ops and Black Mesa Modules for the upcoming sector gameplay. Reverts the remove changes done back on Main Nova Branch in early January-Feberury 2024. Following modules will be placed in modular_nova\modules
> ## How This Contributes To The BlueMoon Roleplay Experience
> 
> Allows further gameplay elements for the upcoming sector where weapons are the main play.
> ## Proof of Testing
> 
> After running in a local server (Pending Changes to the code), confirmed that you are able to purchase Sol Drum Magazines after modifying the Cargo Supply Console Board to "Broad". For balancing issues, currently will keep the standard 60 rounds though can be configured later down the line.
> Screenshots/Videos
> ## Changelog
> 
> 🆑 add: Re-adds Assault Ops and Black Mesa Modules code: Re-enables Qarad Drum Magazine for purchase when supply console board is in "Broad" mode /:cl:

Forgot to reference Novasector main where it got removed, here it is

https://github.com/NovaSector/NovaSector/pull/282/files#diff-e95bbeed7815b1697ece053cae092d2b790844338ccedd0ec9d166aa7726c67d

RS-Chan committed:
Changed 3 files for Drum re-enabled

Edited 3 files that re-enables the drum and by default, is added to the machine gun when you purchase it
